api/models/v0.3/modelBarcode.js

```diff
@@ -70,12 +71,18 @@ class Barcode {
       barcode = result.rows[0].barcode;
       newBarcode = false;
     } catch (error) {
-      // There are no unused barcodes so get the lowest barcode that was
-      // used and subtract one from it. This is the next new available one.
+      // There are no unused barcodes so get the lowest barcode to generate
+      // a new barcode.
       query =
         "select barcode from barcodes where used=true order by barcodes asc limit 1;";
       result = await db.query(query);
-      barcode = `${parseInt(result.rows[0].barcode, 10) - 1}`;
+      // Remove the last digit from the existing barcode since we need a
+      // 13-digit number to pass through the luhn-algorithm.
+      barcode = Math.floor(parseInt(result.rows[0].barcode, 10) / 10);
+      // Subtract one for the next barcode.
+      barcode -= 1;
+      // Pass the value and get a 14-digit barcode.
+      barcode = luhn.calculate(barcode);
```

Comment:
Example of this in the database:
<img width="584" alt="Screen Shot 2020-05-11 at 11 18 31 AM" src="https://user-images.githubusercontent.com/1280564/81580443-72880a00-937b-11ea-8e26-1e4a04348443.png">